playbooks/os/openshift-cluster/config.yml

```diff
@@ -0,0 +1,39 @@
+---
+# TODO: need to figure out a plan for setting hostname, currently the default
+# is localhost, so no hostname value (or public_hostname) value is getting
+# assigned
```

Comment:
Is the metadata not being returned properly or lacking the hostname values?  openshift_facts should be setting the hostname and public_hostname variables based on the instance metadata.  It should only fall back to hostname -f if the metadata does not provide the correct values.